### ISSUE TITLE
How to ignore web3's DeprecationWarning message?

### ISSUE THREAD

**Issue by avatar-lavventura** [ethereum/web3.py#741]
* pip 9.0.3
* Python: 3.5.2
* OS: linux (ubuntu)

When I call my smart-contract's call() I keep receive following message:

`/home/alper/.venv-py3/lib/python3.5/site-packages/web3/utils/decorators.py:14: DeprecationWarning: call is deprecated in favor of contract.<functions/events>.<method name>.call`

[Q] Is there any way to ignore it?

```
### pip install:
$ sudo apt-get install python3-pip
$ sudo pip3 install virtualenv
#once
$ virtualenv -p python3 ~/.venv-py3
# each session:
$ source ~/.venv-py3/bin/activate
```
Current session is: `(.venv-py3) user$`

python file:

```
#!/usr/bin/env python
from __future__ import print_function
from web3 import Web3
#from web3.auto import w3
import json, sys, os
from web3.providers.rpc import HTTPProvider

os.chdir(sys.path[0]);

web3 = Web3(HTTPProvider('http://localhost:8545'))

fileAddr = open("address.json", "r")
contractAddress = fileAddr.read().replace("\n", "")

with open('abi.json', 'r') as abi_definition:
    abi = json.load(abi_definition)

contractAddress = web3.toChecksumAddress(contractAddress);
smartContract  = web3.eth.contract(contractAddress, abi=abi);
array= smartContract.call().getClusterAddresses(); #<=========WARNING message show up.
```

**Comment by dangell7** [comment]
If your smart contract function is getClusterAddresses(), then your code should read smartContract.functions.getClusterAddresses()

**Comment by dangell7** [comment]
https://github.com/samyagency/samy/blob/master/deploy_crowdsale.py 

Check out the function “finalize crowdsale”. 

I use the signed transaction function but you can use the same code up to that point. Make the transaction. transaction = smartContract.functions.yourFunction.buildTransaction(built_transaction) Then use sendTransaction(transaction).

Just make sure if you use my “built_transaction” then change the chain is, nonce, etc

**Comment by dangell7** [comment]
Oops nevermind. Just use smartContract.functions.yourFunction().call()

**Comment by avatar-lavventura** [comment]
It says:
Traceback (most recent call last):
  File "getClusterAddresses.py", line 22, in <module>
    array = smartContract.functions.yourFunction().call();
AttributeError: 'Contract' object has no attribute 'yourFunction' @dangell7

**Comment by avatar-lavventura** [comment]
Your first suggestion works! Cheers.  @dangell7 

Working line:
`smartContract.functions. yourFunction().call();`

**Comment by avatar-lavventura** [comment]
How about for `transact()` calls? @dangell7

**Comment by dangell7** [comment]
I don’t mind helping you, but you need to read the docs...

**Comment by dangell7** [comment]
owner_key is the private key, used for signing the transaction

**Comment by avatar-lavventura** [comment]
I find a easier solution.

Following line suppresses the warning messages. contractCall.py contains `transact()` or `call()` functions.

`python contractCall.py 2>/dev/null`

**Comment by dangell7** [comment]
Right, you don’t have to. Again, you should read the docs. There are 2 types of transactions. Ones that are signed by the private key and ones that are done from an unlocked account. Really the code is the same up until sendTransaction or sendRawTrabsaction.

**Comment by avatar-lavventura** [comment]
Thanks mate, I really appreciated all your help and guide.  @dangell7

**Comment by avatar-lavventura** [comment]
I find a better solution. On `/.venv-py3/lib/python3.5/site-packages/web3/utils/decorators.py` file there is this function:

```
def deprecated_for(replace_message):
    '''
    Decorate a deprecated function, with info about what to use instead, like:

    @deprecated_for("toBytes()")
    def toAscii(arg):
        ...
    '''
    def decorator(to_wrap):
        @functools.wraps(to_wrap)
        def wrapper(*args, **kwargs):
            warnings.simplefilter('always', DeprecationWarning)
            warnings.warn(
                "%s is deprecated in favor of %s" % (to_wrap.__name__, replace_message),
                category=DeprecationWarning,
                stacklevel=2)
            warnings.simplefilter('default', DeprecationWarning)
            return to_wrap(*args, **kwargs)
        return wrapper
    return decorator
```

Line:` warnings.simplefilter('always', DeprecationWarning)` Should be commented out. Also please note that, that line does not exist at https://github.com/ethereum/web3.py/blob/master/web3/utils/decorators.py . @carver @dangell7
